#!/usr/bin/env python
PACKAGE = "dp_core"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("propeller_K_positive", str_t, 0, "K_positive", "0.025, 0.025, 0.025, 0.025")
gen.add("propeller_K_negtive", str_t, 0, "K_negtive", "0.01, 0.01, 0.01, 0.01")
gen.add("propeller_max_rotation", str_t, 0, "max_rotation", "100,100,100,100")
gen.add("propeller_min_rotation", str_t, 0, "min_rotation", "2, 2, 2, 2")
gen.add("propeller_max_alfa", str_t, 0, "max_alfa", "3.32,   0.0174, 3.32, 0.0174 ")
gen.add("propeller_min_alfa", str_t, 0, "min_alfa", "-0.0174,-3.32,  -0.0174,-3.32")
gen.add("propeller_delta_alfa", str_t, 0, "delta_alfa", "0.157, 0.157,0.157,0.157")
gen.add("propeller_delta_rotation", str_t, 0, "delta_rotation", "50,50,50,50")
# gen.add("propeller_max_thrust", str_t, 0, "max_thrust", "100,100,100")
# gen.add("propeller_min_thrust", str_t, 0, "min_thrust", "0,0,0")
gen.add("max_target_force", str_t, 0, "max_target_force", "250,250,250,250")
gen.add("min_target_force", str_t, 0, "min_target_force", "0,0,0,0")
gen.add("allman_switch", bool_t, 0, "allman_switch", True)
gen.add("dp_switch", bool_t, 0, "dp_switch", False)
gen.add("UTMx_switch", bool_t, 0, "UTMx_switch", False)
gen.add("UTMy_switch", bool_t, 0, "UTMy_switch", False)
gen.add("head_switch", bool_t, 0, "head_switch", False)
gen.add("debug_mode", bool_t, 0, "debug_mode", False)


log_enum = gen.enum([ gen.const("info", int_t, 0, "log print flag:INFO"),
                      gen.const("warn", int_t, 1, "log print flag:WARN"),
                      gen.const("error", int_t, 2, "log print flag:ERROR")],
                      "Set Log Level")
gen.add("log_level", int_t, 0, "Set Log Level", 0, 0, 2, edit_method=log_enum)

exit(gen.generate(PACKAGE, "thrustallocation", "thrustallocation"))

